modules/optimization/examples/shape_optimization/forward.i

```diff
@@ -0,0 +1,278 @@
+inner_radius = 6
+outer_radius = 10
+volume_constraint = 200
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+[]
+
+[Mesh]
+  [mesh]
+    type = ConcentricCircleMeshGenerator
+    has_outer_square = no
+    num_sectors = 16
+    radii = '${inner_radius} ${outer_radius}'
+    rings = '16 16'
+    preserve_volumes = false
+  []
+  [inner_radius]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = mesh
+    new_boundary = inner
+    primary_block = 2
+    paired_block = 1
+  []
+
+  [delete]
+    type = BlockDeletionGenerator
+    input = inner_radius
+    block = 1
+  []
+  [gather_all]
+    type = BoundingBoxNodeSetGenerator
+    input = delete
+    bottom_left = '-100 -100 -100'
+    top_right = '100 100 100'
+    new_boundary = total
+  []
+  [combine]
+    type = SideSetsFromBoundingBoxGenerator
+    input = gather_all
+    bottom_left = '-100 -100 -100'
+    top_right = '100 100 100'
+    boundaries_old = 'inner outer'
+    boundary_new = moving
+  []
+
+[]
+[Variables]
+  [disp_x]
+  []
+  [disp_y]
+  []
+  [T]
+  []
+[]
+
+[AuxVariables]
+  [dist_between]
+    [AuxKernel]
+      type = NearestNodeDistanceAux
+      variable = dist_between
+      paired_boundary = moving
+      boundary = total
+      block = 2
+      use_displaced_mesh = false
+      execute_on = "INITIAL"
+    []
+  []
+[]
+
+[Kernels]
+  [disp_x]
+    type = MatDiffusion
+    variable = disp_x
+    use_displaced_mesh = false
+    diffusivity = diff_coef
+  []
+  [disp_y]
+    type = MatDiffusion
+    variable = disp_y
+    use_displaced_mesh = false
+    diffusivity = diff_coef
+  []
+  # run physics of interest on deformed mesh
+  [Diffusion]
+    type = FunctionDiffusion
+    variable = T
+    use_displaced_mesh = true
+  []
+  [Source]
+    type = BodyForce
+    variable = T
+    value = 1
+    use_displaced_mesh = true
+  []
+[]
+
+[Materials]
+  # perserve elements near the boundary
+  [diff_coef]
+    type = ParsedMaterial
+    coupled_variables = 'dist_between'
+    expression = '1/(dist_between+0.5)'
+    property_name = 'diff_coef'
+  []
+  [h]
+    type = ADGenericFunctionMaterial
+    prop_names = h
+    prop_values = h
```

Comment:
Can you add syntax checking testing to the examples? (I take it that regular regression testing may not be possible since they may take longer).

Reply:
Wait... You do have a gold file there, but I can't see `tests`.

Reply:
Nice catch. I have a test for shape optimization that I also use for the shape opt example. This example is no longer needed and I got rid of it.